### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by yutongz**
PR template to help adding release-note

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+**Release note**:
+<!--  Steps to write your release note:
+1. Use the release-note-* labels to set the release note state (if you have access)
```
Comment (ldemailly) [inline .github/PULL_REQUEST_TEMPLATE.md]: I don't understand that one ?

Comment (yutongz) [inline .github/PULL_REQUEST_TEMPLATE.md]: So there are two way to set release-note related labels for mungegithub. One, if you have access to set label directly, just do it. Or if someone don't have access to do it or you want to add more details in release-note, then pr creator can add release message to pr description using this template. Prow will automatically add "release-note" label. If you add "None", "release-note-none" label will be added. Take a look at this [pr](https://github.com/istio/test-infra/pull/356), will help understand what's happening. If you don't do any of these two, "release-note-needed" will be added to block merging.

Comment (ldemailly) [inline .github/PULL_REQUEST_TEMPLATE.md]: Thx! I wanted to ask that during the demo, what does release-note-none mean/is it a good thing ? (my guess is it means skip the PR in release notes ? can you confirm ? so should be the exception right ?)

also what are the other release-note-* we envision (beside none and needed)

Comment (yutongz) [inline .github/PULL_REQUEST_TEMPLATE.md]: Yes, you are right. So totally, we have "release-note" "release-note-none" and "release-note-needed". When people do release, just select all PRs with "release-note" label and you get every release notes. If we don't care about this PR in release just add "release-note-none" or leave "None". "needed" label is the way to enforce people to do this basically :)

**Comment by yutongz** [comment]
@ldemailly  Could you comment "/approve no-issue" and "lgtm" such that mungegithub can take care of this :)

**Comment by ldemailly** [comment]
will the bot auto rebase ? then auto merge ?

**Comment by yutongz** [comment]
The bot did it's job!

**Pull request merged**